orttraining/orttraining/training_ops/cuda/optimizer/gradient_control.cc

```diff
@@ -102,5 +91,65 @@ Status InPlaceAccumulator<T, T_GRAD>::ComputeInternal(OpKernelContext* ctx) cons
   return Status::OK();
 }
 
+#define REGISTER_IN_PLACE_TENSOR_ACCUMULATORV2_TYPED(T, T_GRAD)                       \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(                                                      \
+      InPlaceAccumulatorV2,                                                           \
+      kMSDomain,                                                                      \
+      1,                                                                              \
+      T##_##T_GRAD,                                                                   \
+      kCudaExecutionProvider,                                                         \
+      (*KernelDefBuilder::Create())                                                   \
+          .Alias(0, 1)                              /* Accumulate tensors in-place */ \
+          .InputMemoryType(OrtMemTypeCPUInput, 2)   /* overwrite_flag is on CPU*/     \
+          .OutputMemoryType(OrtMemTypeCPUOutput, 0) /* updated_flag is on CPU*/       \
+          .TypeConstraint("T", DataTypeImpl::GetTensorType<T>())                      \
+          .TypeConstraint("T_GRAD", DataTypeImpl::GetTensorType<T_GRAD>()),           \
+      InPlaceAccumulatorV2<T, T_GRAD>);
+
+REGISTER_IN_PLACE_TENSOR_ACCUMULATORV2_TYPED(float, float)
+
+template <typename T, typename T_GRAD>
+Status InPlaceAccumulatorV2<T, T_GRAD>::ComputeInternal(OpKernelContext* ctx) const {
+  typedef typename ToCudaType<T>::MappedType CudaT;
+  typedef typename ToCudaType<T_GRAD>::MappedType CudaT_GRAD;
+
+  Tensor& left_addee_buffer = *const_cast<Tensor*>(ctx->Input<Tensor>(0));
+  const Tensor& right_addee_buffer = *ctx->Input<Tensor>(1);
+  const Tensor* overwrite_tensor = ctx->Input<Tensor>(2);
+  const bool overwrite = overwrite_tensor != nullptr ? *(overwrite_tensor->template Data<bool>()) : false;
+
+  if (overwrite) {
+    const void* source = right_addee_buffer.template Data<T_GRAD>();
```

Comment:
Do you want to use `const T_GRAD*` here instead? The return type is a `const T*`, so we can take advantage of that.